WordPress/Classes/ViewRelated/Post/AbstractPostListViewController.swift

```diff
@@ -647,6 +666,99 @@ class AbstractPostListViewController : UIViewController, WPContentSyncHelperDele
         presentViewController(navController, animated: true, completion: nil)
     }
 
+    // MARK: - Searching
+
+    func toggleSearch() {
+        searchController.active = !searchController.active
+    }
+
+    func heightForSearchWrapperView() -> CGFloat {
+
+        guard let navigationController = navigationController else {
+            return SearchWrapperViewMinHeight
+        }
+
+        let navBar = navigationController.navigationBar
+        let height = navBar.frame.height + self.topLayoutGuide.length
+
+        return max(height, SearchWrapperViewMinHeight)
+    }
+
+    func isSearching() -> Bool {
+        return searchController.active && currentSearchTerm()?.characters.count > 0
+    }
+
+    func currentSearchTerm() -> String? {
+        return searchController.searchBar.text
+    }
+
+    func updateForLocalPostsMatchingSearchText() {
+        resetTableViewContentOffset()
+        updateAndPerformFetchRequest()
+        refreshCachedRowHeightsForTableViewWidth()
+        tableView.reloadData()
+
+        let filter = currentPostListFilter()
+        if filter.hasMore && tableViewHandler.resultsController.fetchedObjects?.count == 0 {
+            // If the filter detects there are more posts, but there are none that match the current search
+            // hide the no results view while the upcoming syncPostsMatchingSearchText() may in fact load results.
+            hideNoResultsView()
+            postListFooterView.hidden = true
+        } else {
+            refreshResults()
+        }
+    }
+
+    func isSyncingPostsWithSearch() -> Bool {
+        return searchesSyncing > 0
+    }
+
+    func postsSyncWithSearchDidBegin() {
+        searchesSyncing += 1
+        postListFooterView.showSpinner(true)
+        postListFooterView.hidden = false
+    }
+
+    func postsSyncWithSearchEnded() {
+        searchesSyncing -= 1
```

Comment:
Wondering if we could ever end up with < 0 value.  I'm not seeing a way to do it and can't break when I test so maybe its safe, but maybe add an assert for paranoia?